### PR TITLE
HPCC-30771 ECL Watch v9 WUs list filter allow multiple clusters

### PR DESCRIPTION
**Pull request by jeclrsg**
Changed our AsyncDropdown component to accept a "multiSelect" parameter (which is already built-in to the extended FluentUI Dropdown)

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [ ] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: esp/src/src-react/components/forms/Fields.tsx [hunk 1]
```diff
@@ -101,43 +103,93 @@ const AsyncDropdown: React.FunctionComponent<AsyncDropdownProps> = ({
...
+            }
+            if (!items.length) return;
+            if (JSON.stringify(items.map(item => item.key)) === JSON.stringify(selectedKey.split("|"))) {
+                // do nothing, unless
+                if (!selectedItems.length) {
+                    setSelectedItems(items);
+                }
+            } else {
+                setSelectedItems(items);
+            }
```
Comment (GordonSmith) [inline esp/src/src-react/components/forms/Fields.tsx:135]: Lines 127 -> 135 is equivalent to:
```ts
if (items.length && (!selectedItems.length || JSON.stringify(items.map(item => item.key)) !== JSON.stringify(selectedKey.split("|")))) {
   setSelectedItems(items);
}
```
Is that correct (I am not understanding the `!selectedItems.length` condition)?

File: esp/src/src-react/components/forms/Fields.tsx [hunk 2]
```diff
@@ -101,43 +103,93 @@ const AsyncDropdown: React.FunctionComponent<AsyncDropdownProps> = ({
...
+    }, [multiSelect, selectedKey, selOptions, selectedItem, selectedItems]);
 
     React.useEffect(() => {
-        if (!selectedItem || selectedItem?.key === selectedKey) return;
-        if (selectedItem !== undefined) {
-            onChange(undefined, selectedItem, selectedIdx);
+        if (multiSelect) {
+            if (!selectedItems.length) return;
+            if (selectedItems.length > 1) {
+                if (JSON.stringify(selectedItems.map(item => item.key)) === JSON.stringify(selectedKey.split("|"))) return;
```
Comment (GordonSmith) [inline esp/src/src-react/components/forms/Fields.tsx:161]: Minor:  The following would be more efficient I suspect (same for above):
```ts
if (selectedItems.map(item => item.key).join("|") === selectedKey) return;
```

**Review by GordonSmith — CHANGES_REQUESTED** [review]

File: esp/src/src-react/components/forms/Fields.tsx (hunk 1 above)
Comment (GordonSmith) [reply esp/src/src-react/components/forms/Fields.tsx:125]: @jeclrsg bump?

Comment (GordonSmith) [reply esp/src/src-react/components/forms/Fields.tsx:125]: I did have a `!==` in there.  But it was more for me to understand why the `!selectedItem.length` was specialised - probably to do with initial load?

Comment (jeclrsg) [reply esp/src/src-react/components/forms/Fields.tsx:125]: That's right, it's for the initial loading of selectedItems from the url

**Review by GordonSmith — APPROVED** [review]

**Pull request merged**